### PR TITLE
Update phenotype hive import - e104

### PR DESCRIPTION
**Pull request by ima23**
- fix required for GWAS e104 37 import: commit 'skip empty phenotype gwas'
- remove caching  of db adaptor
- update default dir: ENSVAR-3745
- review input ids, don't use overflow: ENSVAR-3666
- add source specific check: ENSVAR-3752
- remove old import before fresh import: ENSVAR-3746
- save genome assembly in internal production db: ENSVAR-3917

### CODE REVIEW

File: modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/CheckPhenotypeAnnotation.pm [hunk 1]
```diff
@@ -377,16 +380,38 @@ sub check_source {
...
+    }
 
-    if (defined $previous_counts->{$check_name} &&
-        $previous_counts->{$check_name}  > $new_counts->{$check_name}){
-      $text_out.= "WARNING: ".$new_counts->{"$table\_count"}." $table entries";
-      $text_out.= " (previously ".$previous_counts->{"$table\_count"}.")" if defined  $previous_counts->{"$table\_count"} ;
+    my $check_name =$check_names[0];
+    if (defined $previous_counts->{phenotype_feature_count_details}{$check_name} &&
+        $previous_counts->{phenotype_feature_count_details}{$check_name}  > $new_counts->{phenotype_feature_count_details}{$check_name}){
+      $text_out.= "WARNING: ".$new_counts->{phenotype_feature_count_details}{$check_name}." $check_name entries";
```
Comment (helensch) [inline modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/CheckPhenotypeAnnotation.pm:396]: Suggest  space between  variable and assignment operator  .=

**Review by helensch — COMMENTED** [review]
@ima23 looks good, just a couple of queries re initialisation and branches we can talk through

File: modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportOMIA.pm [hunk 2]
```diff
@@ -131,7 +131,11 @@ sub fetch_input {
   my $file_omia = 'omia_gene_table.txt';
 
   my $workdir = $pipeline_dir."/".$source_info{source_name_short}."/".$species;
-  make_path($workdir) or die "Failed to create $workdir $!\n";
+  unless (-d $workdir) {
```
[the remark this answers is not in the document]
Comment (ima23) [reply modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportOMIA.pm:134]: I have created this ticket and will reviewed and update any others for it ENSVAR-3961.

File: modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/PhenotypeAnnotation_conf.pm [hunk 3]
```diff
@@ -173,7 +173,7 @@ sub pipeline_analyses {
                 @common_params,
                 run_type => $self->o('run_type'),
             },
-            -input_ids  => [{}],
+            -input_ids  => [], #default
```
[the remark this answers is not in the document]
Comment (ima23) [reply modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/PhenotypeAnnotation_conf.pm:176]: Yes, the pipeline is no longer seeded with one job on initialisation. Confluence command is updated.

File: modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/CheckPhenotypeAnnotation.pm [hunk 4]
```diff
@@ -377,16 +380,38 @@ sub check_source {
...
+      $text_out.= " (previously ".$previous_counts->{phenotype_feature_count_details}{$check_name}.")" ;
       $text_out.= "\n";
-      $count_ok = 0;
+      #for grch37 do not fail the job as prev counts are retrieved by species and not by assembly and grch38 are always reported
```
[the remark this answers is not in the document]
Comment (ima23) [reply modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/CheckPhenotypeAnnotation.pm:399]: As we discussed we still want to see the warning but we don't want the workflow to stop being carried further for this case.

File: modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/PhenotypeAnnotation_conf.pm [hunk 5]
```diff
@@ -203,7 +203,7 @@ sub pipeline_analyses {
                 '5->A' => [ 'import_mimmorbid' ],
                 '6->A' => [ 'import_ddg2p' ],
                 '7->A' => [ 'import_cancerGC' ],
-                'A->1' => [ 'check_phenotypes'],
+                'A->8' => [ 'check_phenotypes'],
```
[the remark this answers is not in the document]
Comment (ima23) [reply modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/PhenotypeAnnotation_conf.pm:206]: Yes, the final job is on branch 8, ImportHuman line 104.

File: modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/PhenotypeAnnotation_conf.pm [hunk 6]
```diff
@@ -374,13 +374,13 @@ sub pipeline_analyses {
...
             },
-            -input_ids      => [],
+            -input_ids      => [], #default
             -hive_capacity  => 1,
             -rc_name    => 'default',
             -flow_into  => {
                 '2->A' => [ 'import_impc'],
                 '3->A' => [ 'import_mgi'],
-                'A->1' => [ 'check_phenotypes'],
+                'A->4' => [ 'check_phenotypes'],
```
[the remark this answers is not in the document]
Comment (ima23) [reply modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/PhenotypeAnnotation_conf.pm:383]: As we discussed, the final job will have the parameters sent on branch 4, ImportMouse line 111.

**Comment by ima23** [comment]
Thanks for feedback @helensch. As we discussed I've made the changes relating to the caching and PR is ready for review again. Thanks.

**Comment by helensch** [comment]
merged into master and release/104
@ima23 I approved the PR, but accidently missed the approve step before merging

**Pull request merged**